[SPARK-12760] [DOCS] invalid lambda expression in python example for …

mortada opened the pull request:
…local vs cluster

@srowen thanks for the PR at https://github.com/apache/spark/pull/10866! sorry it took me a while.

This is related to https://github.com/apache/spark/pull/10866, basically the assignment in the lambda expression in the python example is actually invalid

```
In [1]: data = [1, 2, 3, 4, 5]
In [2]: counter = 0
In [3]: rdd = sc.parallelize(data)
In [4]: rdd.foreach(lambda x: counter += x)
  File "<ipython-input-4-fcb86c182bad>", line 1
    rdd.foreach(lambda x: counter += x)
                                   ^
SyntaxError: invalid syntax
```

mortada committed:
[SPARK-12760] [DOCS] invalid lambda expression in python example for local vs cluster

srowen commented:
Does it still execute without error on a cluster? (even if it doesn't actually increment the counter in the way someone might expect.) Certainly if it doesn't compile we need to change this, but want to make sure the result with "global" executes too.

AmplabJenkins commented:
Can one of the admins verify this patch?

srowen commented:
Jenkins test this please

mortada commented:
@srowen it compiles for local, let me test that on a cluster

I noticed that the next line is actually also invalid python

```
In [7]: print("Counter value: " + counter)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-e340457a6af8> in <module>()
----> 1 print("Counter value: " + counter)

TypeError: Can't convert 'int' object to str implicitly
```

I just updated the PR

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/49880/
Test FAILed.

mortada commented:
@srowen I tested the python code in cluster mode (5 ec2 workers) and this works fine

```
16/01/21 17:33:29 INFO BlockManagerMasterEndpoint: Registering block manager 172.31.10.56:35937 with 6.6 GB RAM, BlockManagerId(4, 172.31.10.56, 35937)
16/01/21 17:33:29 INFO BlockManagerMasterEndpoint: Registering block manager 172.31.10.55:59871 with 6.6 GB RAM, BlockManagerId(0, 172.31.10.55, 59871)
16/01/21 17:33:29 INFO BlockManagerMasterEndpoint: Registering block manager 172.31.10.53:39162 with 6.6 GB RAM, BlockManagerId(1, 172.31.10.53, 39162)
16/01/21 17:33:29 INFO BlockManagerMasterEndpoint: Registering block manager 172.31.10.54:59145 with 6.6 GB RAM, BlockManagerId(2, 172.31.10.54, 59145)
16/01/21 17:33:29 INFO BlockManagerMasterEndpoint: Registering block manager 172.31.10.57:35000 with 6.6 GB RAM, BlockManagerId(3, 172.31.10.57, 35000)
In [1]: data = [1, 2, 3, 4, 5]

In [2]: counter = 0

In [3]: rdd = sc.parallelize(data)

In [4]: def increment_counter(x):
    global counter
    counter += x
   ...:

In [5]: rdd.foreach(increment_counter)
16/01/21 17:34:26 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.31.10.55:59871 (size: 3.2 KB, free: 6.6 GB)
16/01/21 17:34:26 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.31.10.56:35937 (size: 3.2 KB, free: 6.6 GB)
16/01/21 17:34:26 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.31.10.57:35000 (size: 3.2 KB, free: 6.6 GB)
16/01/21 17:34:26 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.31.10.53:39162 (size: 3.2 KB, free: 6.6 GB)
16/01/21 17:34:26 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.31.10.54:59145 (size: 3.2 KB, free: 6.6 GB)
(other output skipped)

In [6]: print("Counter value: ", counter)
Counter value:  0
```

srowen commented:
LGTM then. I can merge this with the other doc updates

srowen commented:
Jenkins test this please

SparkQA commented:
**[Test build #49919 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/49919/consoleFull)** for PR 10867 at commit [`2e1c016`](https://github.com/apache/spark/commit/2e1c0161f8506bccae619e5aae62a69cd16ebd13).

SparkQA commented:
**[Test build #49919 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/49919/consoleFull)** for PR 10867 at commit [`2e1c016`](https://github.com/apache/spark/commit/2e1c0161f8506bccae619e5aae62a69cd16ebd13).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/49919/
Test PASSed.

srowen commented:
Merged to master/1.6
